trino/serial.go

```diff
@@ -35,6 +35,55 @@ func (e UnsupportedArgError) Error() string {
 // If another string format is used it will error to serialise
 type Numeric string
 
+// Date indicates we want a Date type in Trino from a Golang time.
+type trinoDate struct {
+	year  int
+	month time.Month
+	day   int
+}
+
+func Date(year int, month time.Month, day int) trinoDate {
+	return trinoDate{year, month, day}
+}
+
+// Time indicates we want a Time type in Trino from a Golang time.
+type trinoTime struct {
+	hour       int
+	minute     int
+	second     int
+	nanosecond int
+}
+
+func Time(hour int,
```

Comment:
Same here. Also, I wonder if it would be too much to describe what to do when a higher precision is needed.
```suggestion
// Time creates a representation of a Trino Time type. To represent time with precision higher than nanoseconds, pass the value as a string and use a cast in the query.
func Time(hour int,
```